src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -80,6 +83,28 @@ private static byte[] EncodeLength(int length)
         }
 
         /// <summary>
+        /// Encode the segments { tag, length, value } of a boolean.
+        /// </summary>
+        /// <param name="value">The boolean to encode</param>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeBoolean(bool value)
+        {
+            // BER says FALSE is zero, TRUE is other.
+            // DER says TRUE is 0xFF.
+            byte[] data =
+            {
+                (byte)(value ? 0xFF : 0x00),
+            };
+
+            return new byte[][]
+            {
+                new byte[] { (byte)DerSequenceReader.DerTag.Boolean }, 
+                new byte[] { 0x01 }, 
```

Comment:
>Having to build structures like this suggests a stronger type really could be worthwhile, especially so that we can guarantee tags and lengths are right everywhere

Seconded. (though it’d be ok to address in a separate PR as this one’s quite substantial already.)